x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/Aggregations.java

```diff
@@ -29,7 +29,8 @@ private Aggregations() {}
         VALUE_COUNT("value_count", "long"),
         MAX("max", null),
         MIN("min", null),
-        SUM("sum", null);
+        SUM("sum", null),
+        SCRIPTED_METRIC("scripted_metric", null);
```

Comment:
The 1st argument is the name known by aggregations, the 2nd the output type, which is either the concrete type or `null` if the source type should be used. So `null` is a magic variable and actually it should not be null for `scripted_magic`, but that's due to the lack of another magic.

`null` is a bad magic anyway and the types are well defined and limited. So I suggest to change this to:

```
enum AggregationType {
        AVG("avg", "double"),
        CARDINALITY("cardinality", "long"),
        VALUE_COUNT("value_count", "long"),
        MAX("max", "source"),
        MIN("min", "source"),
        SUM("sum", "source"),
        SCRIPTED_METRIC("scripted_metric", "dynamic");
```

There should never be a type `source` and `dynamic`, alternatively we could use `_source`, `_dynamic` to make the special meaning more visible.

Reply:
The method `resolveTargetMapping` needs to be changed, it should return the concrete source type or `dynamic`.

Reply:
Ah, nice, for sure! Let me crank that out today. It would be better than implicitly having different behavior for `null` between `max` and `scripted_metric`.